client/src/Components/HOC/public.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from "react";
+import { Redirect } from 'react-router-dom';
+import checkLogin from "../helper/checkLogin";
+import Home from '../Layouts/Home';
+
+export default WrappedComponent =>
+  class extends Component {
+    state = {
+      fetch: false,
+      login: null,
+      error: null
+    };
+
+    componentDidMount() {
+        checkLogin()
+        .then(response => {
+          const { result } = response;
+          this.setState({ login: result, fetch: true });
+        })
+        .catch(error => {
+          this.setState({ error, fetch: true });
+        });
+    }
+
+    render() {
+      const { fetch, login, error } = this.state;
+      if (!fetch) {
+        return <h1> Loading </h1>;
+      }
+      if(fetch && error) return <p>{error}</p>
+      if (login) return <Redirect to={<Home />} />;
```

Comment:
i just was wondering what's the different between private and public  route (regard to the entire code i didn't check that out ). 
but just a note instead of make a new component called private you can pass a prop to determine if the route was private or public like :

```
 render() {
      const { fetch, login, error } = this.state;
      const { isPrivate } = this.props;
if (!fetch) {
        return <h1> Loading </h1>;
      }
      if (fetch && error) return <p>{error}</p>;
     
 if(isPrivate) {

 if (login) return <WrappedComponent {...this.props} />;
      return <Redirect to={<Login />} />;
    }
} else {
      if (login) return <Redirect to={<Home />} />;
      return <WrappedComponent {...this.props} />;
}
```

just basic login you can make it better . 
Thanks 👍